### ISSUE TITLE
Electron rebuild shouldn't rebuild devDependencies

### ISSUE THREAD

**Issue by MarshallOfSound** [electron/electron-rebuild#96]
Currently (as far as I can tell) `electron-rebuild` rebuilds `devDependencies`.

We can change this by passing in a list of package names to `npm rebuild` when we spawn it in `rebuildNativeModules`.

I've almost got a working branch with this change however I was wondering how people want to handle `optionalDependencies`

For instance I currently have an optional dependency that is technically a dev dependency but it's optional so it can fail.

TLDR: I want to change `electron-rebuild` so it doesn't rebuild devDeps, what should it do for optDeps

/cc @paulcbetts @zeke ?
